client/src/components/Menu/filter.tsx

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable camelcase */
+import React from 'react';
+import { Button, ButtonGroup, Grid } from '@mui/material';
+import { styled } from '@mui/system';
+
+interface Category {
+  id: number;
+  name: string;
+}
+
+interface FilterComponentProps {
+  categories: Category[];
+  onCategorySelect: (category_id: number) => void;
+}
+
+const StyledButton = styled(Button)`
+  color: black;
+  font-family: 'Poppins', sans-serif;
+  background-color: #f3f1f1;
+  border: none;
+  text-transform: none;
+  font-size: 16px;
+  border-radius: 4px;
+  padding: 10px;
+  margin-left: 10px;
+
+  &:hover {
+    background-color: #fe724c;
+    border: none;
+  }
+
+  &:active {
+    background-color: #fe724c;
+    border: none;
+  }
+`;
+
```

Comment:
Professional advice, you need to use the components of either style or emotion with `sx`. Don't use both.